### PR TITLE
feat(table): adding tdsClick event to table rows

### PR DESCRIPTION
**Pull request by ckrook**
## **Describe pull-request**  
This pull-request is regarding adding an tdsClick event to the table rows handle events on a row click.

## **Issue Linking:**  
_Choose one of the following options_
- **GitHub:** [Include issue link  ](https://github.com/scania-digital-design-system/tegel/issues/672)
- **No issue:** Adding a tdsClick event to the table rows

## **How to test**  
_Provide detailed steps for testing, including any necessary setup._
1. Go to storybook and navigate to the table component > basic row.
2. Check in the console that the event is being triggered upon clicking a row.

## **Checklist before submission**
- [x] All existing tests pass
- [x] I have updated the documentation (if applicable)

## **Suggested test steps**
- [x] Browser testing (Chrome, Safari, Firefox) 
- [x] Events

### CODE REVIEW

File: packages/core/src/components/table/table-body-row/table-body-row.tsx [hunk 1]
```diff
@@ -61,6 +64,18 @@ export class TdsTableBodyRow {
...
+  /** Event emitted when a row is clicked. */
+  @Event({
+    eventName: 'tdsClick',
+    composed: true,
+    cancelable: false,
+    bubbles: true,
+  })
+  tdsClick: EventEmitter<{
+    tableId: string;
+    rowIndex: number;
```
Comment (mJarsater) [inline packages/core/src/components/table/table-body-row/table-body-row.tsx:76]: might wanna return the click event here aswell.

File: packages/core/src/components/table/table-body-row/table-body-row.tsx [hunk 2]
```diff
@@ -98,12 +129,16 @@ export class TdsTableBodyRow {
...
       <Host
+        tabindex={this.clickable ? '0' : null}
         class={{
           'tds-table__row': true,
           'tds-table__row--selected': this.selected,
           'tds-table__compact': this.compactDesign,
           'tds-table--divider': this.verticalDividers,
+          'tds-table__row--clickable': this.clickable,
         }}
+        onClick={() => this.handleRowClick()}
```
Comment (mJarsater) [inline packages/core/src/components/table/table-body-row/table-body-row.tsx:140]:
```suggestion
        onClick={(e) => this.handleRowClick(e)}
```

Comment (mJarsater) [reply packages/core/src/components/table/table-body-row/table-body-row.tsx:140]: and them pass the event to the emitter

**Review by nathalielindqvist — COMMENTED** [review]
tdsClick returns tableId's as undefined, not sure if this is because there is no id attached to the Table in our example or because of a bug?
Screenshot 2024-10-10 at 13 11 33

**Comment by mJarsater** [comment]
> tdsClick returns tableId's as undefined, not sure if this is because there is no id attached to the Table in our example or because of a bug? Screenshot 2024-10-10 at 13 11 33

The table should be instaniated with a default id:

from table.tsx
```jsx
tableId: string = generateUniqueId();
```

So it seems like this is some sort of bug.

File: packages/core/src/components/table/table-body-row/table-body-row.tsx (hunk 2 above)
Comment (ckrook) [reply packages/core/src/components/table/table-body-row/table-body-row.tsx]: Thank you for the feedback @mJarsater  👍

File: packages/core/src/components/table/table-body-row/table-body-row.tsx (hunk 1 above)
Comment (ckrook) [reply packages/core/src/components/table/table-body-row/table-body-row.tsx:78]: Thanks! 👍

**Comment by ckrook** [comment]
I fixed the things that @mJarsater suggested. I do get the id now when clicking the rows. @nathalielindqvist do you still experience this issue?

tdsClick-event

**Review by nathalielindqvist — COMMENTED** [review]
Now the tableId's return an id! But the rows are clickable when "Clickable rows" is set to false, but when it's set to true I get an error and the event is not fired:
Screenshot 2024-10-10 at 16 50 44

**Comment by ckrook** [comment]
@nathalielindqvist I fixed the script so this error is not getting triggered. :)

File: packages/core/src/components/table/table-body-row/table-body-row.tsx [hunk 3]
```diff
@@ -70,6 +87,24 @@ export class TdsTableBodyRow {
...
+    this.tdsClick.emit({
+      event: e,
+      tableId: this.tableId,
+      rowIndex: rowIndex,
+    });
+  }
+
+  handleKeyDown(e) {
+    if (this.clickable && (e.key === 'Enter' || e.key === ' ')) {
+      e.preventDefault();
```
Comment (mJarsater) [inline packages/core/src/components/table/table-body-row/table-body-row.tsx:103]: Why is this needed?

Comment (timrombergjakobsson) [reply packages/core/src/components/table/table-body-row/table-body-row.tsx:103]: Good point @ckrook!

Comment (mJarsater) [reply packages/core/src/components/table/table-body-row/table-body-row.tsx:103]: I was more thinking of the `e.preventDefault()`, why is that part needed?

Comment (ckrook) [reply packages/core/src/components/table/table-body-row/table-body-row.tsx:103]: The purpose of e.preventDefault() here is to prevent any default behavior associated with the Enter key or space, which could potentially trigger unintended actions in the browser (like submitting a form or scroll the page)

**Review by theJohnnyMe — APPROVED** [review]
Works and looks good to me.

**Review by timrombergjakobsson — APPROVED** [review]
Nice work @ckrook!

File: packages/core/src/components/table/table-body-row/table-body-row.tsx (hunk 3 above)
Comment (mJarsater) [reply packages/core/src/components/table/table-body-row/table-body-row.tsx:103]: Smart! I'd maybe just leave a comment stating this, just for the future. Feels like a thing that could easily be wrongfully removed in a refactor 🚀

**Review by nathalielindqvist — APPROVED** [review]
Awesome work! 💯
